lib/src/Uncertainty/Algorithm/Sensitivity/HSICEstimatorGlobalSensitivity.cxx

```diff
@@ -0,0 +1,201 @@
+
+    const Point nullDiag(n_);
+    B.setDiagonal(nullDiag, 0);
+
+    const Scalar mHSIC = (1 + Ex * Ey - Ex - Ey) / n_;
+    const Scalar factor = 2.0 * (n_ - 4) * (n_ - 5) / n_ / (n_ - 1) / (n_ - 2) / (n_ - 3) / n_ / (n_ - 1);
+    const Scalar varHSIC = B.computeSumElements() * factor;
+
+    const Scalar alpha = mHSIC * mHSIC / varHSIC;
+    const Scalar beta = n_ * varHSIC / mHSIC;
+
+    const Gamma distribution(alpha, 1.0 / beta);
+    const Scalar p = estimatorType_.computePValue(distribution, n_, HSICobs, mHSIC);
+    PValuesAsymptotic_[dim] = p;
+  }
+  isAlreadyComputedPValuesAsymptotic_ = true ;
+}
+
+/* Get the asymptotic p-values */
+Point HSICEstimatorGlobalSensitivity::getPValuesAsymptotic() const
+{
+  if(!(isAlreadyComputedPValuesAsymptotic_))
+  {
+    computePValuesAsymptotic();
+    isAlreadyComputedPValuesAsymptotic_ = true ;
+  }
+  return PValuesAsymptotic_;
+}
+
+/* Draw the asymptotic p-values */
+Graph HSICEstimatorGlobalSensitivity::drawPValuesAsymptotic() const
+{
+  return drawValues(getPValuesAsymptotic(), "Asymptotic p-values");
+}
+
+/* Compute all indices at once */
+void HSICEstimatorGlobalSensitivity::run() const
+{
+  /* Compute the HSIC and R2-HSIC indices */
+  if(!(isAlreadyComputedIndices_))
+  {
+    computeIndices();
+    isAlreadyComputedIndices_ = true ;
+  }
+
+  /* Compute the p-values by permutation */
+  if(!(isAlreadyComputedPValuesPermutation_))
+  {
+    computePValuesPermutation();
+    isAlreadyComputedPValuesPermutation_ = true ;
+  }
+
+  /* Compute the p-values asymptotically */
+  if(!(isAlreadyComputedPValuesAsymptotic_))
+  {
+    computePValuesAsymptotic();
+    isAlreadyComputedPValuesAsymptotic_ = true ;
+  }
+
+}
+
+/* Method save() stores the object through the StorageManager */
+void HSICEstimatorGlobalSensitivity::save(Advocate & adv) const
+{
+  PersistentObject::save(adv);
+  adv.saveAttribute( "covarianceList_", covarianceList_ );
+  adv.saveAttribute( "inputSample_", inputSample_ );
+  adv.saveAttribute( "outputSample_", outputSample_ );
+  adv.saveAttribute( "estimatorType_", estimatorType_ );
+  adv.saveAttribute( "weightFunction_", weightFunction_ );
+  adv.saveAttribute( "n_", n_ );
+  adv.saveAttribute( "inputDimension_", inputDimension_ );
+  adv.saveAttribute( "HSIC_XY_", HSIC_XY_ );
+  adv.saveAttribute( "HSIC_XX_", HSIC_XX_ );
+  adv.saveAttribute( "HSIC_YY_", HSIC_YY_ );
+  adv.saveAttribute( "R2HSICIndices_", R2HSICIndices_ );
+  adv.saveAttribute( "PValuesPermutation_", PValuesPermutation_ );
+  adv.saveAttribute( "permutationSize_", permutationSize_ );
+  adv.saveAttribute( "isAlreadyComputedIndices_", isAlreadyComputedIndices_ );
+  adv.saveAttribute( "isAlreadyComputedPValuesPermutation_", isAlreadyComputedPValuesPermutation_ );
+  adv.saveAttribute( "PValuesAsymptotic_", PValuesAsymptotic_ );
+  adv.saveAttribute( "isAlreadyComputedPValuesAsymptotic_", isAlreadyComputedPValuesAsymptotic_ );
+}
+
+/* Method load() reloads the object from the StorageManager */
```

Comment:
Same remark here: call the base class load() method then load only the members defined in the class, not in the base class.

Reply:
OK done